src/components/fields/submit-button/submit-button.tsx

```diff
@@ -24,11 +29,22 @@ export const SubmitButton = (props: IGenericFieldProps<ISubmitButtonSchema>) =>
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, []);
 
+	useDeepCompareEffect(() => {
+		if (disabled === "invalid-form") {
+			try {
+				hardValidationSchema.validateSync(formValues);
```

Comment:
I might be mixing it up with formik but isn't there a way to derive the current form state from the form context?

Reply:
yeah I found that it is possible to check via `isValid` in `useFormContext` but it is a getter and appears to be making updates under-the-hood. I found out because it is throwing a bunch of ` act(...)` warnings in the test cases.

so in view of that, I think manually validating the schema is better and faster.